lib/iris/analysis/__init__.py

```diff
@@ -1248,6 +1259,25 @@ def _sum(array, **kwargs):
     return rvalue
 
 
+def _lazy_sum(array, axis, **kwargs):
```

Comment:
I'm curious: why are we duplicating all of this logic, and not just implementing it once lazily? If we want a non-lazy form, we can just compute the result (as in, ``lambda _sum: _lazy_sum(*args, **kwargs).compute()``), right?